modules/KalturaSupport/resources/mw.kAnalony.js

```diff
@@ -433,30 +490,71 @@
 			_this.startTime = null;
 			_this.kClient = mw.kApiGetPartnerClient( _this.embedPlayer.kwidgetid );
 			_this.monitorIntervalObj.cancel = false;
+
 			if ( _this.firstPlay ){
 				_this.sendAnalytics(playerEvent.VIEW, {
 					playTimeSum: _this.playTimeSum,
                     averageBitrate: _this.rateHandler.getAverage(),
-					bufferTimeSum: _this.bufferTimeSum
+					bufferTimeSum: _this.bufferTimeSum,
+                    maxChunkDownloadTime: _this.stats.maxChunkDownloadTime,
+                    chunksDownloaded: _this.stats.chunksDownloaded,
                 });
 				_this.firstPlay = false;
 			}
 			_this.smartSetInterval(function(){
-                if ( !_this._p100Once || (_this.embedPlayer.donePlayingCount > 0)){ // since we report 100% at 99%, we don't want any "VIEW" reports after that (FEC-5269)
-					_this.sendAnalytics(playerEvent.VIEW, {
+				if ( !_this._p100Once || (_this.embedPlayer.donePlayingCount > 0)){ // since we report 100% at 99%, we don't want any "VIEW" reports after that (FEC-5269)
+				try{
+						_this.stats.bufferLength = _this.getPlayer().getCurrentBufferLength();
+					}catch(e){
+					}
+					var dataObj = {
                         playTimeSum: _this.playTimeSum,
                         averageBitrate: _this.rateHandler.getAverage(),
-                        bufferTimeSum: _this.bufferTimeSum
-                    });
+                        bufferTimeSum: _this.bufferTimeSum,
+                        maxChunkDownloadTime: _this.stats.maxChunkDownloadTime,
+						chunksDownloaded: _this.stats.chunksDownloaded,
+					};
+					
+					if(_this.stats.bufferLength >= 0){
+						dataObj.bufferLength = _this.stats.bufferLength
+					}
+
+					_this.sendAnalytics(playerEvent.VIEW, dataObj);
 					_this.bufferTime = 0;
 				}
+				_this.stats.chunksDownloaded=0;
+				_this.stats.bufferLength = -1;
+				_this.stats.maxChunkDownloadTime=-1;
+
 				if ( !_this.monitorViewEvents ){
 					_this.stopViewTracking();
 				}
 			},_this.reportingInterval,_this.monitorIntervalObj);
 
 		},
 
+		setCookie:function(name,value,days) {
+			var expires = "";
+			if (days) {
```

Comment:
1.we never  pass expiry, so i think it will expire after refresh
2.with new cookie policy, i think we will have hard problem with cookie expiry as they now browser doesn
t support unlimited expiry
3. i think we need to consider localStorage, and if we are IE10 not to enable this feature and not send userId